### ISSUE TITLE
ServiceBus - Receive a stream of messages - not working - issue

### ISSUE THREAD

**Issue by kumarrajnumber** [Azure/azure-sdk-for-java#12258]
**Describe the bug**
Service bus sdk not receiving stream messages. Process is stopped without any action.
This is working(batch):
`https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/servicebus/azure-messaging-servicebus#receive-a-batch-of-messages`

With the same connectionstring and queuename, this is not working(stream):
`https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/servicebus/azure-messaging-servicebus#receive-a-stream-of-messages`

***Exception or Stack Trace***
No exceptions

**To Reproduce**
Used same connectionstring and queuename in batch fetching. It's working perfectly fine. But stream message are not working.

***Code Snippet***
```
import com.azure.messaging.servicebus.ServiceBusClientBuilder;
import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
import com.azure.messaging.servicebus.ServiceBusReceiverAsyncClient;
import reactor.core.Disposable;

import java.nio.charset.StandardCharsets;

public class SBConsumer {
    public static void main(String[] args) {

        ServiceBusReceiverAsyncClient receiver = new ServiceBusClientBuilder()
                .connectionString("<<connection string>>")
                .receiver()
                .queueName("myqueue")
                .buildAsyncClient();

        // receive() operation continuously fetches messages until the subscription is disposed.
        // The stream is infinite, and completes when the subscription or receiver is closed.
        Disposable subscription = receiver.receive().subscribe(context -> {
            ServiceBusReceivedMessage message = context.getMessage();
            System.out.printf("Id: %s%n", message.getMessageId());
            System.out.printf("Contents: %s%n", new String(message.getBody(), StandardCharsets.UTF_8));
        }, error -> {
            System.err.println("Error occurred while receiving messages: " + error);
        }, () -> {
            System.out.println("Finished receiving messages.");
        });

        System.out.println(subscription.isDisposed());


    }
}
```

console logs (no errors):
```
[main] INFO com.azure.messaging.servicebus.implementation.ServiceBusConnectionProcessor - namespace[] entityPath[mysb.servicebus.windows.net]: Setting next AMQP channel.
[main] INFO com.azure.messaging.servicebus.implementation.ServiceBusConnectionProcessor - namespace[] entityPath[mysb.servicebus.windows.net]: Next AMQP channel received, updating 0 current subscribers
[main] INFO com.azure.messaging.servicebus.ServiceBusClientBuilder - # of open clients with shared connection: 1
[main] INFO com.azure.messaging.servicebus.ServiceBusReceiverAsyncClient - myqueue: Creating consumer for link 'myqueue_127152_1592305006811'
[main] INFO com.azure.messaging.servicebus.implementation.ServiceBusReceiveLinkProcessor - Requesting a new AmqpReceiveLink from upstream.
[main] INFO com.azure.core.amqp.implementation.ReactorConnection - connectionId[MF_cb61cd_1592305006689]: Creating and starting connection to mysb.servicebus.windows.net:5671
[main] INFO com.azure.core.amqp.implementation.ReactorExecutor - connectionId[MF_cb61cd_1592305006689], message[Starting reactor.]
[main] INFO com.azure.core.amqp.implementation.AzureTokenManagerProvider - Creating new token manager for audience[amqp://mysb.servicebus.windows.net/myqueue], resource[myqueue]
[single-1] INFO com.azure.core.amqp.implementation.handler.ConnectionHandler - onConnectionInit hostname[mysb.servicebus.windows.net], connectionId[MF_cb61cd_1592305006689]
[single-1] INFO com.azure.core.amqp.implementation.handler.ReactorHandler - connectionId[MF_cb61cd_1592305006689] reactor.onReactorInit
[single-1] INFO com.azure.core.amqp.implementation.handler.ConnectionHandler - onConnectionLocalOpen hostname[mysb.servicebus.windows.net:5671], connectionId[MF_cb61cd_1592305006689], errorCondition[null], errorDescription[null]
[main] INFO com.azure.messaging.servicebus.implementation.ServiceBusMessageProcessor - Back-pressure request: 9223372036854775807
false

Process finished with exit code 0
```

**Expected behavior**
I am expecting to receive messages from ServiceBus continuously fetches messages until the subscription is disposed. The same credentials are working with Batch process. But i need stream processing. process should never stop, it should keep fetching the data forever. I need this : "The asynchronous ServiceBusReceiverAsyncClient continuously fetches messages until the subscription is disposed."

**Setup (please complete the following information):**
 - OS: Windows10
 - IDE : IntelliJ 2020.1
 - Version of the Library used:
```
        <dependency>
            <groupId>com.azure</groupId>
            <artifactId>azure-messaging-servicebus</artifactId>
            <version>7.0.0-beta.3</version>
        </dependency>
```

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by joshfree** [comment]
@yvgopal @hemanttanwar @conniey PTAL

**Comment by conniey** [comment]
Thanks for reporting this @kumarrajnumber. `.subscribe` is a non-blocking operation. After it sets up the subscription, it will move onto the next program step, `System.out.println(subscription.isDisposed());` and finally exit the program. If you add a `Thread.sleep(10000)` after the subscribe step, it'll stop the program from exiting.

**Comment by axisc** [comment]
@kumarrajnumber, do let us know if @conniey's response helped unblock you. 
If not, please feel free to open another issue.
